### ISSUE TITLE
Get the trace from the UI

### ISSUE THREAD

**Issue by hmedina** [Kappa-Dev/KaSim#381]
One should be able to request the trace from the UI, specially if running WebSim locally.

Suggestion: put an "export trace" button next to the "run" button on the UI (or in some menu), and if clicked, the UI asks for a filename (e.g. with a save-as dialog box) to stream / save to.

**Comment by hmedina** [comment]
Option to save trace is there, under Simulation Options, but when checked, what should happen? I ran a model but I don't see how to obtain the trace file. It isn't listed under Outputs, and Firefox didn't show the 'save as' dialog...

**Comment by pirbo** [comment]
You run a simulation and pause it (or wait up to the pause condition).
You should see, in the control panel between the "continue" and the "clear" button, a button "get_trace". You click on it, you wait ... you wait ... you wait and firefox should open a popup to offer you to save the file "trace.json"

**Comment by hmedina** [comment]
Ah, but if my model uses `$STOP` and I run until `[FALSE]`? For example:

```
%var: 'Γeq' 1.0e-2
%var: 'γon' 1.0e-4
%var: 'γoff' 'γon' / 'Γeq'
%agent: A(n,m)
'r' A(n),A(m) <-> A(n!1),A(m!1) @ 'γon' {0},'γoff'
%init: 1.0e2 A()
%mod: [T] > 0.5 * 1/([min] 'γon' 'γoff') do ( $SNAPSHOT "t_05_snap.ka" )
%mod: [T] > 1.0 * 1/([min] 'γon' 'γoff') do ( $SNAPSHOT "t_10_snap.ka" )
%mod: [T] > 1.5 * 1/([min] 'γon' 'γoff') do ( $SNAPSHOT "t_15_snap.ka" )
%mod: [T] > 10 * 1/([min] 'γon' 'γoff') do ( $SNAPSHOT "t_100_snap.ka"; $STOP )
```

**Comment by pirbo** [comment]
You wait until it stops and you see "get trace".
Works_For_Me©

**Comment by hmedina** [comment]
I don't see the `get trace` button. Unless the button appears after I wait? I ran this 20m ago (by GitHub comment timestamps), and it still hasn't shown up.

![capture](https://cloud.githubusercontent.com/assets/1588553/26178798/e2b795b0-3b2d-11e7-90bd-86117757c7d5.PNG)

**Comment by hmedina** [comment]
I did the same on Chrome, waited 5 minutes, also no button.

**Comment by pirbo** [comment]
You're right on my local build it works on dev.executableknowledge.org/try/ it does not !!!!

**Comment by hmedina** [comment]
On both I'm using native WebSim. Does that mean the UI not handling properly what WebSim is sending it because it's out of date (e.g. yesterday's build) ?

**Comment by pirbo** [comment]
Yes with yesterday's websim, you won't be able to get the trace as it is a new Rest API call but there is an issue somewhere else in pure javascript (which is enough for your example) I don't see either the "get trace" button whereas I should !

**Comment by pirbo** [comment]
OK I've got it:
```
[master 3d38665] Sync website for js with Kappa-Dev/KaSim@d286c5d51a9568091f047d17641ccdd70bf9251a The trace from the UI (Closes #381)
 4 files changed, 56715 insertions(+), 55397 deletions(-)
 rewrite try/JsSim.js (71%)
 rewrite try/JsSim.js.bak (71%)
 rewrite try/WebWorker.js (67%)
remote: Invalid username or password.
fatal: Authentication failed for 'https://[secure]:@github.com/Kappa-Dev/Kappa-Dev.github.io.git/'
No Changes
```

**Comment by pirbo** [comment]
Pushes from travis to github are back. dev.executableknowledge.org is updated...
